src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs

```diff
@@ -440,6 +440,61 @@ public void TestMixedMultiLineCommentTerminators_01(char outsideDelimiter, char
             Assert.Equal(SyntaxKind.MultiLineCommentTrivia, trivia[0].Kind());
         }
 
+        [Fact]
+        [Trait("Feature", "Comments")]
+        public void TestAtColonTreatedAsComment_RazorRecovery()
+        {
+            var text = "@: More text";
+            var token = LexToken(text);
+
+            Assert.NotEqual(default, token);
+            Assert.Equal(SyntaxKind.EndOfFileToken, token.Kind());
+            Assert.Equal(text, token.ToFullString());
+            var errors = token.Errors();
+            errors.Verify(
+                // error CS1056: Unexpected character '@'
+                TestBase.Diagnostic(ErrorCode.ERR_UnexpectedCharacter).WithArguments("@").WithLocation(1, 1));
+            var trivia = token.GetLeadingTrivia().ToArray();
+            Assert.Equal(1, trivia.Length);
+            Assert.NotEqual(default, trivia[0]);
+            Assert.Equal(SyntaxKind.SingleLineCommentTrivia, trivia[0].Kind());
+        }
+
+        [Fact]
+        [Trait("Feature", "Comments")]
+        public void TestAtColonTreatedAsCommentAsTrailingTrivia_RazorRecovery()
+        {
+            var text = """
+                Identifier @: More text
```

Comment:
i wouldn't mind test cases for those bizarror cases you mentioned (like `@: /*` and this showing up in `#if debug`).  Just to show what the expected behavior is.